### PR TITLE
fix: allow unwanted without selector

### PR DESCRIPTION
**Pull request by andremacola**
If `selector` object is not present in the `queryRules`, the `unwanted` filter is ignored

### CODE REVIEW

File: src/utils/extractWithSelector.js [hunk 1]
```diff
@@ -19,25 +19,31 @@ const countWord = (text) => {
...
+
+      if (parts.length) {
+        return parts
+          .reduce((prev, curr) => prev.concat([curr]), [])
+          .filter((sect) => stripTags(sect).length > MIN_TEXT_LENGTH)
+          .join('')
+      }
     }
 
+    // console.log('text', document.documentElement.innerHTML)
```
Comment (SettingDust) [inline src/utils/extractWithSelector.js:46]:
```suggestion
```

File: src/utils/extractWithSelector.js [hunk 2]
```diff
@@ -19,25 +19,31 @@ const countWord = (text) => {
...
-  if (!selector) return null
   try {
     const document = new DOMParser().parseFromString(html, 'text/html')
-    for (const exclusion of exclusions) {
-      document.querySelectorAll(exclusion).forEach(node => node.remove())
+
+    if (exclusions.length) {
+      for (const exclusion of exclusions) {
+        document.querySelectorAll(exclusion).forEach(node => node.remove())
+      }
```
Comment (SettingDust) [inline src/utils/extractWithSelector.js:28]:
```suggestion
    for (const exclusion of exclusions) {
      document.querySelectorAll(exclusion).forEach(node => node.remove())
    }
```

**Review by SettingDust — CHANGES_REQUESTED** [review]

File: src/utils/extractWithSelector.js (hunk 1 above)
Comment (andremacola) [reply src/utils/extractWithSelector.js:46]: oh, sorry for that

File: src/utils/extractWithSelector.js [hunk 3]
```diff
@@ -19,23 +19,27 @@ const countWord = (text) => {
...
-      const text = node.innerHTML.trim()
-      if (countWord(text) >= MIN_SECTION_LENGTH) { parts.push(text) }
-    })
-
-    if (parts.length) {
-      return parts
-        .reduce((prev, curr) => prev.concat([curr]), [])
-        .filter((sect) => stripTags(sect).length > MIN_TEXT_LENGTH)
-        .join('')
+    }
```
Comment (SettingDust) [inline src/utils/extractWithSelector.js:28]:
```suggestion
```

**Review by SettingDust — CHANGES_REQUESTED** [review]

**Review by SettingDust — APPROVED** [review]

**Comment by ndaidong** [comment]
@andremacola thank you, it seems there is some points need to be fixed, could you help to verify the test scripts?

**Comment by ndaidong** [comment]
@andremacola thank you, it seems there is some points need to be fixed, could you help to verify the test scripts?

**Comment by andremacola** [comment]
> @andremacola thank you, it seems there is some points need to be fixed, could you help to verify the test scripts?

I'm looking into.

**Comment by andremacola** [comment]
Maybe we need to change some tests to cover try-catch blocks

**Comment by andremacola** [comment]
I had to use `parseHTML` from likedom because `document.documentElement.innerHTML` was not returning all html string from document

File: src/utils/stripUnwantedTags.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+/**
+ * @param html {string}
+ * @param exclusions {string[]}
+ * @returns {string}
+ */
+export default (html, exclusions = []) => {
+  if (typeof exclusions !== 'undefined' && !exclusions.length) return html
+
+  const { document } = parseHTML(html)
```
Comment (SettingDust) [inline src/utils/stripUnwantedTags.js:13]: Use the `new DOMParser().parseFromString(html, 'text/html')`. `DOMParser` will be the native class when running in the browser.
https://github.com/ndaidong/article-parser/blob/main/src/browser/linkedom.js
The polyfills of linkedom only export the `DOMParser` now

File: src/utils/stripUnwantedTags.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+import { parseHTML } from 'linkedom'
+
+/**
+ * @param html {string}
+ * @param exclusions {string[]}
+ * @returns {string}
+ */
+export default (html, exclusions = []) => {
+  if (typeof exclusions !== 'undefined' && !exclusions.length) return html
```
Comment (SettingDust) [inline src/utils/stripUnwantedTags.js:11]:
```suggestion
  if (!exclusions?.length) return html
```

Take advantage of modern syntax

File: src/browser/linkedom.js [hunk 6]
```diff
@@ -1 +1,2 @@
 export const DOMParser = global.DOMParser
+export const parseHTML = global.parseHTML
```
Comment (SettingDust) [inline src/browser/linkedom.js:2]: There isn't a `parseHTML` in browser.You need provide that or use `DOMParser`

**Comment by SettingDust** [comment]
> I had to use parseHTML from likedom because document.documentElement.innerHTML was not returning all html string from document

Need explain

**Review by SettingDust — CHANGES_REQUESTED** [review]

File: src/browser/linkedom.js (hunk 6 above)
Comment (andremacola) [reply src/browser/linkedom.js]: My bad. It was a conflict with my project

**Review by SettingDust — APPROVED** [review]

**Pull request merged**